### PR TITLE
Move UI functions a seperate file

### PR DESCRIPTION
**Pull request by thomaspj10**
* Move the ui code into `ui.js`

No gameplay should be affected. The main file has gotten ~2x smaller when this PR is merged. This will allow me to further focus on rewriting/removing dumb code to make adding new features easier!

### CODE REVIEW

File: js/ui.js [hunk 1]
```diff
@@ -0,0 +1,586 @@
...
+        2: "small",
+        3: "medium",
+        4: "large",
+        5: "x-large",
+        6: "xx-large",
+        7: "xxx-large",
+    }
+
+    if (id < 0) id = 0
+    if (id > 7) id = 7
```
Comment (James103) [inline js/ui.js:401]: This could be rewritten in one line as follows:
```js
id = Math.max(0, Math.min(7, id))
```

File: js/ui.js [hunk 2]
```diff
@@ -0,0 +1,586 @@
...
+    document.getElementById("essenceGainDisplay").textContent = format(getEssenceGain())
+    document.getElementById("essenceGainButtonDisplay").textContent = "+" + format(getEssenceGain())
+
+    // TODO Why don't we use getGameSpeed()
+    document.getElementById("timeWarpingDisplay").textContent = "x" + format(
+        gameData.taskData["Time Warping"].getEffect() *
+            gameData.taskData["Temporal Dimension"].getEffect() *
+            gameData.taskData["Time Loop"].getEffect() *
+            (gameData.requirements["Eternal Time"].isCompleted() ? 2 : 1)
+        )
```
Comment (James103) [inline js/ui.js:469]: `getGameSpeed()` returns 0 if age &ge; lifespan or if the game is paused.

The function could be split into two: `getUnpausedGameSpeed()` which is the same as current `getGameSpeed()` but without the `+!gameData.paused` and `+isAlive()` parts, and a new `getGameSpeed()` which just returns `getUnpausedGameSpeed()` if `!gameData.paused && isAlive()`.

The UI code above can just use `getUnpausedGameSpeed()` in place of what is effectively a copy of its content.

File: js/ui.js (hunk 1 above)
Comment (thomaspj10) [reply js/ui.js:401]: I like the previous aproach more.

File: js/ui.js (hunk 2 above)
Comment (thomaspj10) [reply js/ui.js:469]: Agreed. Will implement in a future refactor!

**Review by indomit — APPROVED** [review]

**Pull request merged**